### DISCUSSION TITLE
Geode v2.0.0-beta.1

### DISCUSSION THREAD

**Discussion by matcool** [geode-sdk/sdk#425] [Announcements]
Our first Geode v2.0 beta! 2.2 support and much more.

No MacOS release yet, as no one has done much progress on it.

## v2.0.0-beta.1
 * 2.2 support!
   * Android arm64-v8 support!
   * Future version compatibility mode, where the Geode UI is disabled
 * Split bindings into their own repo (3aa3ae7d)
 * Add `gd` as a required key in mod.json, to specify target GD version (f7a3947c)
 * Load native binaries from mod resources (acd9bd78)
 * Refactor json library, namespace and headers renamed to `matjson` (b9a09f73)
 * Refactor `AsyncWebRequest` class (d56c70fe)
   * `postFields` renamed to `body` and `bodyRaw`
   * `customRequest` renamed to `method`
 * Change api for `CCDictionaryExt`, now 2nd template arg must be a pointer, making it more similar to std::map (e1b7b1f4)
 * Refactor Hook and Patch classes, Mod::addHook renamed to Mod::hook (a5a3a08d)
 * Remove many deprecated keys from `mod.json`, most importantly `required` in dependencies (39d372c7, b4235bb5)
 * Refactor logging, now pipes directly to fmtlib (2ba8761f)
 * Refactor gd::string (24bbc65f)
 * Refactor many other parts of the codebase

There are many other things included in the v1.4.0 changelog. However since that version never officially released, heres their changes:
## v1.4.0
 * Add Android support !!!!
   * Implement every Geode functionality except `utils::file::openFolder`
   * Requires the Geode launcher in order to be used
   * Fixes the text input node allowing typing for 1 less character
   * Uses `logcat` in order to get crash reports, so reopening the game is required to generate them
   * Broma requires classes to be added `[[link(android)]]` in order to be linked
   * All Geode and GD files are stored in ~~`Android/data/com.geode.launcher/files`~~ its in media now
     * Game files in `game`, save files in `save`
 * Allow logging to be disabled per mod (6d599a5)
 * Mod cells use layouts (114fa46)
 * MacOS console is now separate (182984d)
 * Add uninstall button to Geode mod (Only functional in Windows currently) (a738320)
 * Make new version label invisible on download (0f179da)
 * Fix the toggling of disabled dependencies (cd89ef1)
 * Fix spritesheet issues (ef47647)
 * Change `LoadingLayer` (ef47647)
 * Make mod info popup top a layout (dd806e0)
 * Add `GEODE_HIDDEN` to inline unique functions (71a79ab)
 * Fix big mod icons (26a6c7e)
 * Fix `CCNode::removeChildByID` export (23cd456)
 * Make `MDTextArea` fit its size (140f38b)
 * Enable ESC/Back to go back in Geode mod list (2847bee)
 * Add `SimpleTextArea` (7f277a7)
 * Check modified date when unzipping `.geode` files (5c765c6)
 * Only hash markdown files on resource checking (f563c46)

## Pull Requests
* Android ci by @matcool in https://github.com/geode-sdk/geode/pull/246
* Improve Android bindings by @Cvolton in https://github.com/geode-sdk/geode/pull/275
* Added a simple text area UI node by @SMJSGaming in https://github.com/geode-sdk/geode/pull/277
* Fixed alignment issues caused by unallocated memory in textarea by @SMJSGaming in https://github.com/geode-sdk/geode/pull/282
* add android paddings to PlayerObject by @dankmeme01 in https://github.com/geode-sdk/geode/pull/286
* Updated the text area to introduce new features by @SMJSGaming in https://github.com/geode-sdk/geode/pull/291
* Android build action by @matcool in https://github.com/geode-sdk/geode/pull/303
* Bindings: Fix ProfilePage and CommentCell virtuals by @Cvolton in https://github.com/geode-sdk/geode/pull/304
* add virtual CCBlockLayer::customSetup by @dankmeme01 in https://github.com/geode-sdk/geode/pull/306
* Bindings: Fix DailyLevelPage updateTimers calling convention by @Cvolton in https://github.com/geode-sdk/geode/pull/308
* add missing GJRewardTypes by @Cvolton in https://github.com/geode-sdk/geode/pull/309
* Bindings: add SearchButton by @Cvolton in https://github.com/geode-sdk/geode/pull/322
* Update 1.4.0 with main commits by @matcool in https://github.com/geode-sdk/geode/pull/334
* add DailyLevelNode::init by @Cvolton in https://github.com/geode-sdk/geode/pull/367
* Fix Android strip not being run without Geode CLI by @Cvolton in https://github.com/geode-sdk/geode/pull/371
* Bindings: Improve song related stuff by @Cvolton in https://github.com/geode-sdk/geode/pull/377
* 2.0.0 dev by @matcool in https://github.com/geode-sdk/geode/pull/382
* update bindings to use new 2.200 version by @matcool in https://github.com/geode-sdk/geode/pull/383
* Add saturationChecked and brightnessChecked to ccTypes.h by @adafcaefc in https://github.com/geode-sdk/geode/pull/385
* Update some enums to 2.2 by @Cvolton in https://github.com/geode-sdk/geode/pull/394
* Refactor logging internals by @matcool in https://github.com/geode-sdk/geode/pull/386
* Add more 2.2 enum values by @Cvolton in https://github.com/geode-sdk/geode/pull/395
* "Incompatability" 🚫 "Incompatibility" ✅ by @MarioMastr in https://github.com/geode-sdk/geode/pull/403
* Update PlayerButton in Enums.hpp by @makitard in https://github.com/geode-sdk/geode/pull/404
* Some more 2.2 stuff by @Cvolton in https://github.com/geode-sdk/geode/pull/405
* update json library, changes namespace to matjson by @matcool in https://github.com/geode-sdk/geode/pull/399
* add 2.2 SpecialRewardItems by @Cvolton in https://github.com/geode-sdk/geode/pull/412
* use string_view instead of string& in Mod by @dankmeme01 in https://github.com/geode-sdk/geode/pull/417
* Refactor geode::web by @matcool in https://github.com/geode-sdk/geode/pull/410
* fill in LevelLeaderboardMode enum by @Cvolton in https://github.com/geode-sdk/geode/pull/419
* Refactor some stuff in Loader by @cgytrus in https://github.com/geode-sdk/geode/pull/420
* Add 2.204 support by @Cvolton in https://github.com/geode-sdk/geode/pull/422
* Refactor Hook/Patch by @cgytrus in https://github.com/geode-sdk/geode/pull/421
* c++ exception handler improvements + 64-bit compatibility by @dankmeme01 in https://github.com/geode-sdk/geode/pull/423

## New Contributors
* @adafcaefc made their first contribution in https://github.com/geode-sdk/geode/pull/385
* @MarioMastr made their first contribution in https://github.com/geode-sdk/geode/pull/403

@altalk23 @hjfod

**Full Changelog**: https://github.com/geode-sdk/geode/compare/v1.3.10...v2.0.0-beta.1

<hr /><em>This discussion was created from the release <a href='https://github.com/geode-sdk/geode/releases/tag/v2.0.0-beta.1'>Geode v2.0.0-beta.1</a>.</em>

**Comment by Tamarindo-XD** [comment]
holaaaaaa
